frontend/front/src/routing/ProtectedRoute.js

```diff
@@ -0,0 +1,16 @@
+import * as routes from "./routes";
+
+import { Navigate } from "react-router-dom";
+import { useUserHasRoles } from "../features/login/loginUtils";
+
+/*
+ * Redirects user if they do not have the correct credentials
+ */
+export const ProtectedRoute = ({ children, allowedRoles = [] }) => {
+  const userIsAllowed = useUserHasRoles(allowedRoles);
+  if (!userIsAllowed) {
+    // user is not authenticated
+    return <Navigate to={routes.LOGIN_ROUTE} />;
```

Comment:
If the user isn't logged in they'll be redirected to the login page if they try to go to an unauthorized page. If they're logged in but have the wrong role, they'll get redirected to the login page and then the login page will redirect to the approved dashboard page.